### DISCUSSION TITLE
Feature request: Drafts & auto-saving

### DISCUSSION THREAD

**Discussion by richardvanbergen** [payloadcms/payload#136] [Feature Requests & Ideas]
I've been talking to my manager and we're trying to consider payload as a platform to build on top of outside of our own site.

One thing he's concerned about is the lack of a draft function. There's the preview function but the data returned from the API is only what you saved in the document so you must save the document in order for the preview to work.

This isn't as much of a problem for us as our site is static, but I can see issues where edited content that isn't ready to publish yet gets published as a result of another collection being updated and I can also assume not every site ever made will be statically generated.

Also, it would be nice if it auto-saved like the way that sanity does it.

- Is this on your roadmap?
- Do we have an ETA the roadmap you've been talking about publishing for a while?

**Comment by jmikrut** [comment]
Hey @richardvanbergen — we are **actively** working on identifying the best approach for exactly what you're looking for. Alongside of this functionality, we are also incorporating Revisions, which will allow you to compare / restore older versions of documents. 

As for the roadmap itself, that's a good question. We are aiming to launch it within the next 1 to 2 months. You'll definitely see Drafts & Revisions featured, and we can certainly work an `autosave` feature in as well!

Will update this Discussion with more once we have it. This set of features is high up on our priorities list and might even hit before the roadmap itself.

**Reply by richardvanbergen** [reply]
Thank you very much! I know you guys are busy but it's very reassuring that you have similar priorities.

**Comment by yandodov** [comment]
@richardvanbergen just FYI, Payload now has:

- [Drafts feature](https://payloadcms.com/docs/versions/drafts)
- [Autosave feature](https://payloadcms.com/docs/versions/autosave)

@jmikrut you could close this.
